Added Role UI BDD feature file

pcardiga opened the pull request:
BDD scenarios for Role UI (list, view, CRUD) - https://jira.ez.no/browse/EZP-24685

Todo:
- [x] Implementation

Later follow up:
- Validation of Role limitations view with links and human readable names
- Role assignment
- Role versioning
- Role list pagination

pcardiga commented:
ping @andrerom , @miguelcleverti , @micszo 
Please review

micszo commented:
Shouldn't the last scenario before "Deleting Roles" be called "cancel the update of a role"? :)

pcardiga commented:
you are right @micszo , there was a typo there.

andrerom commented on Features/RoleUI/RoleUI.feature:
```diff
@@ -0,0 +1,166 @@
+    And I edit the "Organizers" role
+    And I set "old_Organizers" as the "Name"
+    When I click on "Save" button
+    Then the Role is successfully published
```
these published / not published could also instead verify presence / non presence of the Role list afterwards right? (depends a bit on how @glye  setup redirects)

pcardiga replied:
Yes it can. In BDD discussions apart from the current one, it was decided to leave the "publish" action in separate scenarios from the "view" action.

andrerom replied:
ok

glye replied:
Save/cancel/delete actions redirect to role list, yes.

andrerom commented on Features/RoleUI/RoleUI.feature:
```diff
@@ -0,0 +1,166 @@
+  @javascript @common
+  Scenario: cancel the creation of a role
+    Given I am on the Roles page
+    And I create a new role
```
Is it intentional that role is sometimes not written with capital R?

pcardiga replied:
It was a way to differentiate different conceptual things: the name of a place (the "R|r"oles page) from a name of an object (the role itself). But i see this decision is raising doubts, so it's not as clear as i tough when i wrote the sentences. Being so, it might be better to level everything and use smallness in both cases.

andrerom replied:
Let it be for now, I need to think about it as well :)

andrerom commented on Features/RoleUI/RoleUI.feature:
```diff
@@ -0,0 +1,166 @@
+
+  @javascript @edge
+  Scenario: Display the information of a role with empty policies
+    Given I have an "Organizers" role
```
This sentence is not clear, when I first read it I assumed it meant Administrator had this role (assigned). Because `I am an Administrator`, so `I have <role>` basically means it is assigned to me :)

But what is being implied here from the use is that it exist, priorly created, so maybe rephrase this sentence to more clearly relfect that.

pcardiga replied:
The objective is to have an "Organizers" role as pre-condition for the test action.
Maybe ``` Given an "Organizers" role exists ``` fits better the purpose.

andrerom replied:
yes, something like that is more clear

glye replied:
/me agrees.

andrerom commented on Features/RoleUI/RoleUI.feature:
```diff
@@ -0,0 +1,166 @@
+  Scenario: Delete a role
+    Given I have an "Organizers" role
+    When I delete the "Organizers" role
+    Then the Role is successfully deleted
```
same here as published, could just as well be a sentence verifying it is no longer present in list of roles, would be more clear as to how you expect the change to affect the application.

andrerom commented:
Other then open questions/discussions: +1

glye commented:
+1

pcardiga commented:
Discussed questions added to scenarios:
- Updated sentence ``` Given I have an "Organizers" role ```to ``` Given an "Organizers" role exists ```
- Added a verification sentence in scenarios after the updating/deleting/cancelling actions

andrerom commented:
Ok, so ready for implementation I guess.

In addition to this we need:
- PR on ezplatform adding the new profile / scenarios
- Change travis testing PR to reflect new profile / scenarios to test

And Qa will need to give us heads up when everything is passing locally so we can safly merge all :)

glye commented:
I just realised there is no pagination in the role list. We need that, and BDD for it.

andrerom commented:
true, but different story, moved it to backlog and marked that it will be fixed in eZ Platform.

miguelcleverti commented:
@andrerom implementations added

andrerom commented:
@miguelcleverti this branch is rebased/merged wrong, seems this PR needs to be reopened against master as the upstream PR is merged by now.

miguelcleverti commented:
@andrerom maybe it is better to just colsed this PR and I will create a new branch and cherry-pick the commits and open a new PR?

andrerom commented:
> @andrerom maybe it is better to just colsed this PR and I will create a new branch and cherry-pick the commits and open a new PR?

yes this is the simplest way.

miguelcleverti commented:
New PR #343

pcardiga commented:
Closed in favor of PR [#343](https://github.com/ezsystems/PlatformUIBundle/pull/343)
